### PR TITLE
samples: mpsl: timeslot: thread-safe MPSL timeslot sample

### PR DESCRIPTION
**Pull request by sofu1**
Make MPSL timeslot sample thread-safe by calling all MPSL functions
from cooperative thread. This way, MPSL function calls cannot be
interrupted by mpsl_low_priority_process().

Signed-off-by: Solveig Fure <Solveig.Fure@nordicsemi.no>

### CODE REVIEW

**Comment by sofu1** [comment]
This is mostly to demonstrate how MPSL API should be used. I haven't experienced that the sample crashes without this change, but I think it could crash.

File: samples/mpsl/timeslot/src/main.c [hunk 1]
```diff
@@ -119,30 +130,73 @@ static void mpsl_timeslot_demo(void)
...
 
-	err = mpsl_timeslot_session_close();
+	api_call = CLOSE_SESSION;
+	err = k_msgq_put(&mpsl_api_msgq, &api_call, K_FOREVER);
 	if (err) {
 		error();
 	}
 }
 
+/* MPSL API calls must be done from a non-preemptible thread.
```
Comment (rugeGerritsen) [inline samples/mpsl/timeslot/src/main.c]: What about: "To ensure thread safe operation, call all MPSL APIs from a non-preemtible thread"

Comment (sofu1) [reply samples/mpsl/timeslot/src/main.c]: Ok, I can replace the whole comment with this.

**Comment by joerchan** [comment]
> This is mostly to demonstrate how MPSL API should be used. I haven't experienced that the sample crashes without this change, but I think it could crash.

I think main thread might be preemptible based on Kconfig option, so it could happen.

**Pull request merged**